slurm_ops_manager/singularity.py

```diff
@@ -0,0 +1,64 @@
+"""Singularity Class."""
+
+import logging
+import shlex
+import subprocess
+from pathlib import Path
+
+from ops.framework import Object, StoredState
+from slurm_ops_manager.utils import operating_system
+
+
+logger = logging.getLogger()
+
+
+class Singularity(Object):
+    """Responsible for Singularity operations."""
+
+    _stored = StoredState()
+
+    def __init__(self, parent, key):
+        """Initialize class."""
+        super().__init__(parent, key)
+
+        self._stored.set_default(singularity_installed=False,
+                                 resource_name=str())
+
+        self._operating_system = operating_system()
+
+        # setup resource name based on operating system
+        if self._operating_system == "ubuntu":
+            self._stored.resource_name = "singularity-deb"
+        elif self._operating_system == "centos":
+            self._stored.resource_name = "singularity-rpm"
+        else:
+            logger.error(f"#### Unsupported OS: {self._operating_system}")
+            self._stored.resource_name = ""
+
+    @property
+    def installed(self) -> bool:
+        """Return wether singularity is installed."""
+        return self._stored.singularity_installed
+
+    @property
+    def resource_name(self) -> str:
+        """Return singularity resource name."""
+        return self._stored.resource_name
+
+    def install(self, resource_path: Path):
+        """Install Singularity."""
+
+        logger.debug(f"#### Installing singularity from: {resource_path}")
+
+        cmds = {"ubuntu": f"apt install {resource_path} -y",
+                "centos": f"yum localinstall {resource_path} -y"}
```

Comment:
It is not recommended to run `apt` non-interactively, use `apt-get` instead.

```suggestion
        cmds = {"ubuntu": f"apt-get install --yes {resource_path}",
                "centos": f"yum localinstall --assumeyes {resource_path}"}
```